packages/video_player/elinux/gst_video_player.cc

```diff
@@ -167,6 +170,45 @@ int64_t GstVideoPlayer::GetCurrentPosition() {
   return position / GST_MSECOND;
 }
 
+#ifdef USE_EGL_IMAGE_DMABUF
+void* GstVideoPlayer::GetEGLImage(void* egl_display, void* egl_context) {
+  std::shared_lock<std::shared_mutex> lock(mutex_buffer_);
```

Comment:
Is there any reason to use shared_lock?

Reply:
I think it is necessary for the same reason as `GetFrameBuffer()`.

Reply:
Ah, right.